store/tikv/coprocessor.go

```diff
@@ -306,8 +306,10 @@ const minLogCopTaskTime = 300 * time.Millisecond
 // send the result back.
 func (it *copIterator) work(ctx goctx.Context, taskCh <-chan *copTask) {
 	defer it.wg.Done()
+	childCtx, cancel := goctx.WithCancel(ctx)
```

Comment:
It's necessary to fork a sub context every time `work` is run since it's called like this:

same file, line 347
```
func (it *copIterator) run(ctx goctx.Context) {
	it.wg.Add(it.concurrency)
	// Start it.concurrency number of workers to handle cop requests.
	for i := 0; i < it.concurrency; i++ {
		go it.work(ctx, it.taskCh)
	}
```